aesara/link/numba/dispatch.py

```diff
@@ -283,9 +285,22 @@ def create_tuple_string(x):
 
 @singledispatch
 def numba_typify(data, dtype=None, **kwargs):
+    try:
+        for i, item in enumerate(data):
+            data[i] = numba_typify(item)
```

Comment:
@brandonwillard Is this what you meant by changing `numba_typify` so that it returns the "storage" list ?

Reply:
Yeah, just an implementation that takes/uses and returns a "storage" list, so that we can keep the `RandomState`-specific requirements in `numba_typify_RandomState` (i.e. recreate the "storage" list in _there_).

This particular change makes the primary `data` argument a `list`, though, which means we can't recreate a "storage" list in the dispatch specializations when/if that's needed.  Instead, we can change the first/`data` argument into a `storage` argument that contains the single-element "storage" list (i.e. `state` in `numba_typify_RandomState` will be a list containing the `RandomState` object) and make `numba_typify` return the same list.

Reply:
I don't exactly understand how this will be implemented though.

If you don't mind, can you go ahead and implement what you have in mind in this PR. (Cause the current implementation in this PR will have quite bit of overhead because of looping over each data member.)